File: ScaleformUI_Csharp/Menus/MenuBase.cs
using ScaleformUI.Menu;

namespace ScaleformUI.Menus
{
    public class MenuBase
    {
        private bool visible;
        public virtual bool Visible
        {
            get => visible;
            set
            {
                visible = value;
                MenuHandler.ableToDraw = value;
            }
        }
        public List<InstructionalButton> InstructionalButtons { get; set; }
        internal virtual void ProcessControl(Keys key = Keys.None)
        {

        }
        internal virtual void ProcessMouse()
        {

        }
        internal virtual void Draw()
        {
        }
    }
}

File: ScaleformUI_V8/src/menus/menu.base.ts
export abstract class BaseMenu {
    private accessor visible: boolean = false
    public instructionalButtons: any[] = [] //TODO! type this
    public Items: any[] = [] // no need to type this.. as well.. every menu has its own types...
    abstract processMouse(): void
    abstract processControl(): void
    abstract draw(): void

    public set Visible(state: boolean) {
        this.visible = state
    }
    public get Visible(): boolean {
        return this.visible
    }
}